org.osgi.test.common/src/main/java/org/osgi/test/common/inject/TargetType.java

```diff
@@ -21,29 +22,60 @@ public TargetType(Class<?> type, List<Type> genericType) {
 		this.genericTypes = genericType;
```

Comment:
Should this constructor be private to make sure people use the static `of` methods?